src/Compilers/CSharp/Portable/Binder/SwitchBinder_BindPatternSwitch.cs

```diff
@@ -12,44 +12,179 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal partial class SwitchBinder
+    // We use a subclass of SwitchBinder for the pattern-matching switch statement until we have completed
+    // a totally compatible implementation of switch that also accepts pattern-matching constructs.
+    internal partial class PatternSwitchBinder : SwitchBinder
     {
-        private BoundPatternSwitchStatement BindPatternSwitch(SwitchStatementSyntax node, Binder originalBinder, DiagnosticBag diagnostics)
+        // Even though these fields exist in the base class, we record them again here.
+        // Our ultimate intent is to remove the type SwitchBinder and have the fully functional
+        // switch binding implementation present in this class.
+        private readonly SwitchStatementSyntax _switchSyntax;
+        private TypeSymbol _switchGoverningType;
+        private readonly GeneratedLabelSymbol _breakLabel;
+        private readonly bool _isPatternSwitch;
+
+        internal PatternSwitchBinder(Binder next, SwitchStatementSyntax switchSyntax) : base(next, switchSyntax)
+        {
+            _switchSyntax = switchSyntax;
+            _breakLabel = new GeneratedLabelSymbol("break");
+            _isPatternSwitch = ((CSharpParseOptions)switchSyntax.SyntaxTree.Options).IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching);
```

Comment:
Yes, it will distinguish the semantics we use for binding the switch expression.

---
In reply to: [64953635](https://github.com/dotnet/roslyn/pull/11485#discussion_r64953635) [](ancestors = 64953635)